### PR TITLE
rsmeter: introduce datasink

### PR DESCRIPTION
**Pull request by zhongzc**
### What is changed and how it works?

Part of #11691 

A refactor to make it easier to introduce pubsub next.

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/resource_metering/src/reporter/mod.rs [hunk 1]
```diff
@@ -20,18 +24,18 @@ use tikv_util::worker::{Runnable, RunnableWithTimer, Scheduler};
 /// [Scheduler]: tikv_util::worker::Scheduler
 /// [RawRecords]: crate::model::RawRecords
 /// [Records]: crate::model::Records
-pub struct Reporter<C> {
-    client: C,
+pub struct Reporter<D> {
+    datasink: D,
```
Comment (breeswish) [inline components/resource_metering/src/reporter/mod.rs:28]:
```suggestion
    data_sink: D,
```

**Review by breeswish — APPROVED** [review]

**Comment by breeswish** [comment]
@mornyx PTAL

**Comment by mornyx** [comment]
`components/test_raftstore/src/server.rs#init_resource_metering()`  also need update due to #11085 has been merged.

**Comment by mornyx** [comment]
LGTM

**Review by breeswish — APPROVED** [review]

**Review by andylokandy — APPROVED** [review]

**Pull request merged**